Adds new custom help messages for politeiawwwcli commands

s-ben opened the pull request:
This adds new custom help messages for four politeiawwwcli commands. Per Issue #474 , we want custom help messages for all politeiawwwcli commands.

lukebp commented on politeiawww/cmd/politeiawwwcli/commands/version.go:
```diff
@@ -1,5 +1,21 @@
+None
+
+Result:
+{
```
```
{
  "version":  (string)  Version of backend 
  "route":    (string)  Version route
  "pubkey":   (string)  Server public key
  "testnet":  (bool)    Whether of not testnet is being used
}`
```

s-ben replied:
Done

lukebp commented on Gopkg.lock:
```diff
@@ -3,39 +3,50 @@
 
```
Don't include changes to Gopkg.lock in your PR.

lukebp commented on politeiawww/cmd/politeiawwwcli/commands/censorcomment.go:
```diff
@@ -8,6 +8,28 @@ import (
+2. commentID   (string, required)   Id of the comment
+3. reason      (string, required)   Reason for censoring the comment
+
+Result:
```
The first object that you have listed under `Result` is not actually part of the result.  It is the request body that is being sent up to the server.  Most of the time this is going to be the same as `Arguments`, unless a user signature.  I would do something like this:

```
censorcomment "token" "commentID" "reason"

Censor a user comment.  Requires admin privileges.

Arguments:
1. token       (string, required)   Proposal censorship token
2. commentID   (string, required)   Id of the comment
3. reason      (string, required)   Reason for censoring the comment

Request:
{
  "token":      (string)  Censorship token
  "commentid":  (string)  Id of comment
  "reason":     (string)  Reason for censoring the comment
  "signature":  (string)  Signature of censor comment (Token+CommentID+Reason)
  "publickey":  (string)  Public key used for signature
}

Response:
{
  "receipt":  (string)  Server signature of comment sensor signature
}
```

s-ben replied:
Have applied the `Request:`, `Response:` format to all commands that output the request:

censorcomment
changepassword
changeusername
manageuser
edituser
Newcomment
newuser
setproposalstatus
votecomment

lukebp commented on politeiawww/cmd/politeiawwwcli/commands/editproposal.go:
```diff
@@ -13,6 +13,58 @@ import (
+  "signature": (string)  Signature of the merkle root 
+}
+{
+  "proposal": {
```
Test alignment for field descriptions.

lukebp commented on politeiawww/cmd/politeiawwwcli/commands/editproposal.go:
```diff
@@ -13,6 +13,58 @@ import (
+3. attachmentFiles   (string, optional)   Attachments 
+
+Result:
+{
```
Same comment as above about differentiating between the request body and the response.  Apply this to all the commands that have a request body.

lukebp commented on politeiawww/cmd/politeiawwwcli/commands/censorcomment.go:
```diff
@@ -8,6 +8,28 @@ import (
+// Help message displayed for the command 'politeiawwwcli help censorcomment'
+var CensorCommentCmdHelpMsg = `censorcomment "token" "commentID" "reason"
+
+Censor comment (admin).
```
I would write out what `(admin)` means.  Something like:

`Censor a user comment.  Requires admin privileges.`

Same for all commands that require admin privileges.

s-ben replied:
Put `Requires admin privilages` on all commands that require admin. The only one I wasn't sure of was `edituser`, where I broke it out by bit flag (copied current copy in comment below).

lukebp commented on politeiawww/cmd/politeiawwwcli/commands/edituser.go:
```diff
@@ -4,6 +4,20 @@ import (
+Edit the details for the given user id (admin).
+
+Arguments:
+1. emailnotifications       (uint64, optional)   Whether to notify via emails
```
This description should  be something like:

`Email notifications setting.  Uses bit flags.`

s-ben replied:
To keep within 80 char width, shortened to:

```
Arguments:
1. emailnotifications (uint64, optional)  Email notification setting (bit flag)
```

lukebp commented on politeiawww/cmd/politeiawwwcli/commands/edituser.go:
```diff
@@ -4,6 +4,20 @@ import (
+
+Result:
+{
+  "emailnotifications": null
```
I would set this as a `uint64` in this example.  So something like:

`"emailnotifications": 1,`

s-ben replied:
Have reworded edituser to:

```
Edit the details for the given user id. Requires admin privileges.

Valid options are:
'0' - User does not receive email notifications   
'1' - User receives email notificaions

Arguments:
1. emailnotifications (uint64, optional)  Email notification setting (bit flag)

Result:
{
  "emailnotifications":  (uint64)  Bit flag
}
{}`
```

s-ben replied:
Updated notification options also, per below updated comment.

brunobraga95 commented on politeiawww/cmd/politeiawwwcli/commands/editproposal.go:
```diff
@@ -13,6 +13,58 @@ import (
+      }
+    ],
+    "numcomments":   (uint)    Number of comments on the proposal
+    "version": 		 (string)  Version of proposal
```
could this be in int?

s-ben replied:
I think so. Got that datatype from here,
https://github.com/decred/politeia/blob/f13e494415aa6553be9d965e5b05a1fbc8475284/politeiawww/api/v1/v1.go#L385

lukebp commented on .gitignore:
```diff
@@ -14,3 +14,4 @@ vendor
 *.orig
 *.diff
 *.sw?
+Gopkg.lock
```
Don't add the Gopkg.lock file to .gitignore.  Revert the Gopkg.lock file back to the one found on master and then just don't commit the changes that get made to it.

lukebp commented on politeiad/politeiad.go:
```diff
@@ -1051,4 +1051,4 @@ func main() {
 		os.Exit(1)
 	}
-}
+}
```
Remove.

lukebp commented on politeiawww/backend.go:
```diff
@@ -3416,4 +3416,4 @@ func convertWWWPropCreditFromDatabasePropCredit(credit database.ProposalCredit)
 		TxID:          credit.TxID,
 	}
-}
+}
```
Remove.

lukebp commented on politeiawww/cmd/politeiawwwcli/commands/editproposal.go:
```diff
@@ -131,4 +185,4 @@ func (cmd *EditProposalCmd) Execute(args []string) error {
 	// Print response details
 	return Print(epr, cfg.Verbose, cfg.RawJSON)
-}
+}
```
Remove.

lukebp commented on politeiawww/cmd/politeiawwwcli/commands/edituser.go:
```diff
@@ -4,6 +4,24 @@ import (
+
+Valid bit flags are:
+'0' - User does not receive email notifications   
+'1' - User receives email notificaions
```
This part is not accurate.  The valid vote bits can be found here:

https://github.com/decred/politeia/blob/f13e494415aa6553be9d965e5b05a1fbc8475284/politeiawww/api/v1/v1.go#L223-L231

s-ben replied:
Took a stab at descriptions based on variable names. Also changed the command description because not all vote bits require admin privileges (or do they?). Have this now:

```
Edit the details for the given user id. 

Valid bit flags are:
'0' - Notification email when status of user's proposal changes
'1' - Notification email when voting has started on user's proposal
'2' - Notification email when a proposal is vetted
'3' - Notification email when a proposal is edited
'4' - Notification email when voting on a proposal has started
'5' - Notification email when a new proposal is submitted (admin only)
'6' - Notification email when a user authorizes vote on proposal (admin only)
'7' - Notification email when a comment is made on a user's proposal
'8' - Notification email when a comment is made on a user's comment
```

lukebp replied:
We really should have human readable options for this, but that's for a different PR.  For now, the valid vote bits can be specified like this.

```
Valid bit flags are:
'1 << 0' - Notification email when status of user's proposal changes
'1 << 1' - Notification email when voting has started on user's proposal
'1 << 2' - Notification email when a proposal is vetted
'1 << 3' - Notification email when a proposal is edited
'1 << 4' - Notification email when voting on a proposal has started
'1 << 5' - Notification email when a new proposal is submitted (admin only)
'1 << 6' - Notification email when a user authorizes vote on proposal (admin only)
'1 << 7' - Notification email when a comment is made on a user's proposal
'1 << 8' - Notification email when a comment is made on a user's comment
```

lukebp commented on politeiawww/cmd/politeiawwwcli/politeiawwwcli.conf:
```diff
@@ -0,0 +1,8 @@
+; ------------------------------------------------------------------------------
+
+; host=https://proposals.decred.org/api
+host=https://127.0.0.1:4443
```
Remove.

lukebp commented on politeiawww/paywall.go:
```diff
@@ -632,4 +632,4 @@ func (b *backend) SpendProposalCredit(u *database.User, token string) error {
 	err := b.db.UserUpdate(*u)
 	return err
-}
+}
```
Remove.

s-ben commented:
@lukebp, addressed all feedback. Have left comments pointing out a few things that might need further review/tweaking. Have also replied to comments in my [spreadsheet](https://docs.google.com/spreadsheets/d/1KwM2EqNTof2RUuYB0l1ZO1XBmng_kaWMqcmGZgEN_Y8/edit#gid=0) tracking changes. 

Notes:

- Have reverted back to master on Gopkg.lock, politeiad.go and backend.go. Hopefully those are clean now. 
- Removed votes command help for now. Ran into an error there, and realized it was probably because I need to set up a wallet on testnet, buy tickets, etc. That seems like another big chunk of work, so figured I'd leave it out of this PR and tackle that (and the remaining commands) in another PR.
